### PR TITLE
chore: jest upgrade

### PR DESCRIPTION
**Pull request by dangreen**
##### Checklist

Before taking this PR from the draft, please, make sure you have done the following:

- [x] Pipeline is passed
- [ ] Tests are added (including unit tests and stories in the storybook)
- [x] Tests are passed successfully
- [ ] If you're adding a new component/new props, add stories that describe how this component/prop works
- [x] Changeset(s) is(are) added
- [x] You have passed the threshold of the library size
- [x] Commit message follows [commit guidelines](https://github.com/cube-js/cube-ui-kit/blob/main/CONTRIBUTING.md)

Closes:  N/A 

#

### Other information

Can't upgrade to React 18 because upgrade to @types/react@^18.0.0 causes the following error:

```
Error: src/tasty/tasty.tsx(215,16): error TS2590: Expression produces a union type that is too complex to represent.
```

### CODE REVIEW

File: .lintstagedrc [hunk 1]
```diff
@@ -1,3 +1,3 @@
 {
-    "*.(js|ts|tsx|jsx)": ["prettier --write", "eslint --cache --fix", "pnpm test -- --bail --findRelatedTests"]
+    "*.(js|ts|tsx|jsx)": ["prettier --write", "eslint --cache --fix", "pnpm test -- --bail --findRelatedTests --passWithNoTests"]
```
Comment (MrFlashAccount) [inline .lintstagedrc:2]: By the way, as we touched this file, could you remove running tests on the commit?

**Review by MrFlashAccount — APPROVED** [review]

File: .lintstagedrc (hunk 1 above)
Comment (tenphi) [reply .lintstagedrc:2]: Well, I like running tests on a commit :) Is there an issue with that?

**Review by tenphi — APPROVED** [review]

File: .lintstagedrc (hunk 1 above)
Comment (dangreen) [reply .lintstagedrc:2]: We can move test running to prepush hook.

Comment (MrFlashAccount) [reply .lintstagedrc:2]: @dangreen agree. 

@tenphi Running tests on commit slow down your work. If you want to reduce the feedback loop, run tests while you develop a feature using `--watch` flag.

IMHO, preventing committing "inappropriate" code is a bad practice that slows development and gives nothing in return.

What if you must commit current state of work, which might not be "perfect"? There might be plenty of reasons for that.

**Pull request merged**